packages/datastore/src/textfield.ts

```diff
@@ -641,4 +641,40 @@ namespace Private {
     }
     return false;
   }
+
+  /**
+   * Splice data into an array.
+   *
+   * #### Notes
+   * This is intentionally similar to Array.splice, but chunks the splices into
+   * multiple splices so that it does not crash if the number of spliced IDs
+   * is greater than the maximum number of arguments for a function.
+   *
+   * @param arr - the array on which to perform the splice.
+   *
+   * @param start - the start index for the splice.
+   *
+   * @param deleteCount - how many indices to remove.
+   *
+   * @param items - the items to splice into the array.
+   *
+   * @returns an array of the deleted elements.
+   */
+  function spliceArray<T>(arr: T[], start: number, deleteCount?: number, items?: ReadonlyArray<T>): ReadonlyArray<T> {
+    if (!items) {
+      return arr.splice(start, deleteCount);
+    }
+    let size = 100000;
+    if (items.length < size) {
+      return arr.splice(start, deleteCount || 0, ...items);
+    }
+    let deleted = arr.splice(start, deleteCount);
+    let n = Math.floor(items.length / size);
+    let idx = 0;
+    for (let i = 0; i < n; i++, idx += size) {
+      arr.splice(idx, 0, ...items.slice(idx, size));
```

Comment:
I might be reading this wrong, but shouldn't it be like this?
```suggestion
      arr.splice(start, 0, ...items.slice(idx, size));
```
A brief sanity check which checks that it behaves the same as `splice` might be reassuring.

Reply:
Good catch! It shouldn't be `start` or `idx`, but instead something like `start + idx`.

Reply:
I added a test, but not a very good one :/